### ISSUE TITLE
No audio output

### ISSUE THREAD

**Issue by bartart3d** [musescore/MuseScore#24605]
### Issue type

General playback bug

### Description with steps to reproduce

Using Ubuntustudio 24.04, fresh install and updates, running Wayland and Pipewire:

* download AppImage for Musescore 4.4.1
* Launch Musescore
* Open any file, or create music
* Press Play
* Expected behavior: playback
* Actual behavior: no playback

When launching from terminal this message shows as relevant message:
09:30:58.634 | ERROR | main_thread     | AudioModule::setupAudioDriver | audio output open failed

* Go to preferences, attempt to select audio device.
* After selecting system default, entry field remains grayed out (see screenshot)

### Supporting files, videos and screenshots

![afbeelding](https://github.com/user-attachments/assets/c50642b9-0980-4637-872b-8d84abd87949)

### What is the latest version of MuseScore Studio where this issue is present?

4.4.1.242490810

### Regression

I was unable to check

### Operating system

Ubuntu 24.04

### Additional context

To add to the relevancy and importance of this bug I want to point out that there is not one single fully functional 4.x build of Musescore available for Ubuntu 24.04 (I can't speak of other Ubuntu versions) and this AppImage is the recommended build for linux!

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by MarcSabatella** [comment]
The end result symptom is obviously the same, but given the log message is different, and that much about the internals have changed since 2022, it's probably not the same in terms of the actual cause.  On the other hand, probably very little about the details in that other issue are applicable anymore even if the end result is the same, so I'm notnot sure the other issue is that worth keeping open.  Unless someone can confirm the details do still remain relevant.  I can't reproduce any problem myself on Debian.

For the record, audio *does* work in general on Linux, but there are a number of different audio systems that can possibly be configured by any different distribution and on any given system (ALSA, PulseAudio, PipeWire etc), and of course different audio devices may have their own drivers etc.  On systems where things don't work correctly using all defaults, usually some tweak to the audio configuration will work.  But FWIW, if there were one distribution I'd consider worth going out of our way to be sure we support out of the box, it would be Ubuntu Studio.  So to me this issue is the one most worth tracking.

**Comment by bartart3d** [comment]
I agree with MarcSabatella that the end result is the same, but that the issue is probably different.
I'm not technically skilled enough to find out though. I read through the comments didn't find the error I mentioned.

This Flathub version:" https://flathub.org/apps/org.musescore.MuseScore" has audio output, but does not detect Muse hub and has issues with the GUI (mixer won't show up, and other parts neither) 
But maybe a comparison can be made to see what the difference concerning the audio is between the Flatpak version and the AppImage version.

My assumption is that it should be the same, since the source code should be the same, so maybe it is a packaging issue.
Again, I'm not not familiar with coding itself, but I've seen issues with permissions to access audio devices etc. in other projects. Could it be as simple as Musescore not having the permission to open the audio device? But how do you solve that in a AppImage?

grtz,
Bart

**Comment by sl0m0ZA** [comment]
I have the same issue with this version of Musescore Appimage, Ubuntu 24.04 and Pipewire

**Comment by Skyfrei** [comment]
Issue persists on Arch 4.5.1

**Comment by MarcSabatella** [comment]
Is the terminal error message the same for all of those?  Did you try pipewire-alsa?  Some have reported that works best on some systems.

**Comment by bartart3d** [comment]
It's not a fix, it's a workaround, but yes, I can confirm it as working. 
Wow, now the dev's have something to work on to make an actual fix :-)

**Comment by MarcSabatella** [comment]
Seems like a bug in your audio system if trying to set it to 44.1 kHz would cause it to not function at all - best to report that to the developers of that system.  Meanwhile, if your audio device won't respond at 44.1, you can tell MuseScore to use 48 in Edit / Preferences / Audio & MIDI

**Comment by bartart3d** [comment]
@MarcSabatella: read the initial bug, then compare it with the proposed workaround.

I'll summarize: 
audio in Musescore hangs when the system audio is different from 44.1, changing the rate to 44.1 in the system causes Musescore to play again, it's a fix.
Changing the prefs in Musescore doesn't change a thing because, again, look at the screenshot in initial post!

This is not an issue with the system, it's an issue with Musescore!

**Comment by MarcSabatella** [comment]
The thing is, audio does *not* hang on all systems that have system audio different from 44.1.  It seems to be something unique to certain audio services.  Some work just fine out of the box, others stutter by default but work fine when changing the sample rate within MuseScore.  It seems your audio device won't even let MuseScore connect to it at all, which is highly unusual but there do seem to exist a handful of reports.  But at this point, it's not at all clear how representative this is, or where the fault lie.  Is it that something about how MuseScore makes the request to connect that is invalid and your audio system is correct to refuse the request, or if it that something about your audio device is non-standard and refusing a valid request?  I don't think there is enough info to make that determination yet.  But since MuseScore does connect to other audio services, it seems prudent to report it to the developer of yours.

Your screenshot shows you didn't have the ability to change sample rate within MuseScore previously, but I'm guessing you do know.  That is, if you set your system temporarily to 44.1, then start MuseScore, then change it to 48, then change your system back to 48, then restart MuseScore, that may work.

**Comment by Skyfrei** [comment]
As the previous comment said and as my workaround also noted. Changing the sample rate on musescore just crashes the app, then you default back to 44.1k once the app starts. Having to write to other Developers about a musescore issue makes no sense, and this is a recuring event which also happened in tuxguitar and i Provided the solution in src code as well. Its as simple as giving a choice to sample rate or defaulting to something else. Obviously, this will not work if server has a different sample rate because this is how signal processing works. You cant compare 2 signals with different sample rates, thats why up upscaling exists in the first place.

Either ways, this is a workaround and if you dont consider it a solution could we please have it tagged as such so it can help other people who have been looking for one for years now.

All the best.

**Comment by MarcSabatella** [comment]
The crash is obviously a MuseScore issue, although I don't know that anyone else can reproduce it.  But it's still not clear if the original problem - the inability to connect at all at 44.1 kHz - is an issue with MuseScore or with your audio system.  As mentioned, it does work with most other systems - they either change their sample rates or else there is simply a pitch/speed difference.  So right now it still isn't clear wnhy some systems won't produce audio at all.  I have no involvement with that code, so I can't really say.  I'm just pointing out there is a very good chance there is an issue in the audio system you are using if it behaves differently in this respect than all other audio systems.

**Comment by Skyfrei** [comment]
Im unsure what other audio systems do, but with pipewire this didnt work either, and plain alsa neither. If you have a system thats working then i would assume the audio server you just got lucky and the rates match since 44.1 is the default for almost anything OR the sever maps to the lowest sample rate, in this case musescore and then downscales everything. Or the other way around, it just upscales musescore to everything. But this would be a feature, and a feature i havent seen anywhere in arch or ubuntu audio servers, at least not without huge tweaking.
